plugins/azure-devops-backend/src/service/router.ts

```diff
@@ -216,12 +216,14 @@ export async function createRouter(
       req.query.host?.toString() ?? config.getString('azureDevOps.host');
     const org =
       req.query.org?.toString() ?? config.getString('azureDevOps.organization');
+    const path = req.query.path?.toString() ?? 'README.md';
```

Comment:
let's validate the path here:

```suggestion
    if(typeof req.query.path !== 'string') {
      throw new InputError('Invalid path param');
    }
    const path = req.query.path || 'README.md';
```

perhaps `||` is more correct in this case so that it will protect us in case someone sends an empty path?

Reply:
I can add more validation, though I think this specific change doesn't work - for backwards compatibility, I'd like to accept a missing `path` querystring (i.e. `undefined`) and default that to `README.md`. But, I can make the change to cope with `undefined` and a single string, and throw for anything else that comes out of the query string parsing.

If we're going to be more strict then I think the empty string is bad input, and we should reject it rather than silently replacing the value.

One hesitation - if I add more type validation, it'll be a departure from the parsing of the other querystring parameters in this service. Is it worthwhile to deviate here? If adding more validation is useful, should I also add type checks to the other places we read from the querystring? My inclination is to keep things consistent in this PR, and maybe do the validation as a separate PR - but I'll follow your lead.

Reply:
mmm there is no breaking change in adding a validation to `path`.

> If we're going to be more strict then I think the empty string is bad input, and we should reject it rather than silently replacing the value.

I agree that rejects would be better, but your code right now accepts an empty string, so if you think empty string is a bad input isn't better to replace it with a default value instead of break things? 🤔 

> One hesitation - if I add more type validation, it'll be a departure from the parsing of the other querystring parameters in this service. Is it worthwhile to deviate here? If adding more validation is useful, should I also add type checks to the other places we read from the querystring? My inclination is to keep things consistent in this PR, and maybe do the validation as a separate PR - but I'll follow your lead.

I think the way parameters are parsed here is wrong and risky as there is no validation. For example:
`const top = req.query.top ? Number(req.query.top) : DEFAULT_TOP;` this will produce `NaN` in case `req.query.top` isn't parsable. Ideally any input should be validated and If the existing inputs aren't validated it doesn't mean we should continue making mistakes by skipping validating any new parameter

Reply:
OK. I'll work on making the validation better (for this param, in this PR). And file an issue to improve the validation on the other params / other methods? I think it can be done as part of a separate PR.

Reply:
Done. New behaviour:
* undefined = use `README.md`. This is the old behaviour, and is also a reasonable default. It's what happens if the new annotation is not present.
* something other than a string = return HTTP 400
* empty string (we know this won't work) = return HTTP 400
* a valid-looking string = attempt to use the file. Rely on the AzureDevOps server to sensibly handle missing files (equivalently: incorrect paths) by returning an HTTP 404.

The only place this behaves badly is if you specify a path to a folder (rather than a path to a file). I can't sensibly detect this as part of parsing the input values, so I think the "validation" part of this is done. See my comment earlier for more thoughts on how to deal with folders.

Reply:
nice!